test/test_cuda.py

```diff
@@ -429,7 +535,11 @@ def tmp(self):
         self.assertEqual(cpu_tensor, gpu_tensor, precision)
         self.assertEqual(cpu_args, gpu_args, precision)
         # Compare results
-        self.assertEqual(cpu_result, gpu_result, precision)
+        if fn == 'element_size' and t.__name__ == 'HalfTensor':
+            # Workaround since cpu_result is float
+            self.assertEqual(cpu_result, gpu_result * 2, precision)
```

Comment:
I'm a little confused by the `* 2`. Was there a division earlier in the code?

Reply:
element_size for half is 2, for float is 4. But since element_size is special-cased anyway, I would just do ```self.assertEqual(gpu_result, 2)``` for clarity.

Reply:
No this is simply because we used float as cpu type for half. torch.FloatTensor().element_size()=4 so I put a *2 there. I changed this to `self.assertEqual(2, gpu_result)` to make it more clear.